### ISSUE TITLE
Add Method To Count Pages

### ISSUE THREAD

**Issue by gudenau** [Zequez/FactorioMods#14]
I would like a method that allows me to count the number of pages available, so I do not have to guess while my app is loading the mod list.

**Comment by ApocDev** [comment]
Simplest method around this:

```
                for (int i = 1; i < int.MaxValue; i++)
                {
                    Debug.WriteLine($"Fetching mods page {i}");
                    var modsStr = await wc.DownloadStringTaskAsync($"http://api.factoriomods.com/mods?page={i}");
                    var mods = JsonConvert.DeserializeObject<List<Mod>>(modsStr);

                    if (mods == null || mods.Count == 0)
                    {
                        break;
                    }
```

Would be nice to know the count before hand, but either way, you end up with 1 extra REST request. (Note: an empty page will just return "[]" if that's more helpful for your language of choice)

**Comment by Zequez** [comment]
What I usually do is check if the number of mods returned by the page are less than the requested (25 by default IIRC). I'll be adding pagination data to the headers in the near future (in the Python port), so you can read it directly from there.

**Comment by ApocDev** [comment]
I prefer to avoid "magic numbers" in any code whenever possible. If for some reason the API starts returning 20 instead of 25, your method breaks down.

Having proper metadata would be nice.

**Comment by Zequez** [comment]
@ApocDev I agree with you, sadly the current release doesn't let you specify the number of mods, it's a fixed 20 (I just checked). In the Python port this changes (default 25, up to 100 per page), and you also get the headers so you won't need it anyway. For now you can just check if the next page returns no mods.
